### PR TITLE
GEODE-8098: Refactor RedisSetCommandsFunctionExecutor

### PR DESCRIPTION
**Pull request by prettyClouds**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/CommandFunction.java [hunk 1]
```diff
@@ -50,152 +47,115 @@ public static void register() {
...
+  protected Object compute(Region localRegion, ByteArrayWrapper key,
+      RedisCommandType command, Object[] args) {
+    Callable<Object> callable;
     switch (command) {
       case SADD: {
         ArrayList<ByteArrayWrapper> membersToAdd = (ArrayList<ByteArrayWrapper>) args[1];
-        stripedExecutor.execute(key,
-            () -> RedisSet.sadd(localRegion, key, membersToAdd),
-            (addedCount) -> resultSender.lastResult(addedCount));
+        callable = () -> new Long(RedisSet.sadd(localRegion, key, membersToAdd));
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/CommandFunction.java:82]: I think Long.valueOf is preferred over new Long. It has the benefit of returning a cached instance of Longs in the range of -128..127 which is probably what we have most of the time in this code.

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/RedisSet.java [hunk 2]
```diff
@@ -79,14 +79,18 @@ public static long sadd(Region<ByteArrayWrapper, RedisSet> region,
     }
   }
 
-  public static long srem(Region<ByteArrayWrapper, RedisSet> region,
+  public static Long srem(Region<ByteArrayWrapper, RedisSet> region,
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/RedisSet.java:82]: The old code that returned "long" was okay wasn't it? Any reason to change this to Long?

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/RedisSetCommandsFunctionExecutor.java [hunk 3]
```diff
@@ -50,73 +47,57 @@ public RedisSetCommandsFunctionExecutor(Region<ByteArrayWrapper, RedisSet> regio
...
-    return results.getResult().get(0);
+    return CommandFunction.execute(SSCAN, key, new Object[] {matchPattern, count, cursor}, region);
   }
 
-  private ResultCollector executeFunction(RedisCommandType command,
-      ByteArrayWrapper key,
-      Object commandArguments) {
-    return CommandFunction.execute(region, command, key, commandArguments);
-  }
+
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/RedisSetCommandsFunctionExecutor.java:101]: remove these blank lines if they are not needed

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/SingleResultCollector.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+package org.apache.geode.redis.internal.executor.set;
+
+import java.util.concurrent.TimeUnit;
+
+import org.apache.geode.cache.execute.FunctionException;
+import org.apache.geode.cache.execute.ResultCollector;
+import org.apache.geode.distributed.DistributedMember;
+
+public class SingleResultCollector<T> implements ResultCollector<T, T> {
+  T result;
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/SingleResultCollector.java:25]: make this field private?

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/SingleResultCollector.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Override
+  public T getResult(long timeout, TimeUnit unit)
+      throws FunctionException {
+    return result;
+  }
+
+  @Override
+  public void addResult(DistributedMember memberID, T resultOfSingleExecution) {
+    result = resultOfSingleExecution;
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/SingleResultCollector.java:40]: Should this check to see if result != null and throw an IllegalStateException? If someone accidently uses this and gives it multiple results it seems like an exception would be more helpful than just having the last one win

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/StripedExecutor.java [hunk 6]
```diff
@@ -35,9 +34,7 @@
    *
    * @param stripeId defines the "stripe"
    * @param callable the unit of work to do sequentially. May be called after run returns.
-   * @param resultConsumer is given the result of the callable.
    */
-  public <T> void execute(Object stripeId,
-      Callable<T> callable,
-      Consumer<T> resultConsumer);
+  public <T> T execute(Object stripeId,
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/StripedExecutor.java:38]: remove the javadocs on this method that talk about "the result of..."

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/CommandFunction.java (hunk 1 above)
Comment (prettyClouds) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/executor/CommandFunction.java]: yea, this is just a relic of me trying to debug stuff and trying dumb ideas...i removed the boxing.

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/SingleResultCollector.java (hunk 5 above)
Comment (prettyClouds) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/SingleResultCollector.java:46]: yea, that feels much better.  done

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/StripedExecutor.java (hunk 6 above)
Comment (prettyClouds) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/StripedExecutor.java:36]: done.

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/RedisSet.java (hunk 2 above)
Comment (prettyClouds) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/RedisSet.java]: no good reason...reverted.

**Pull request merged**